node/src/accountant/payable_dao.rs

```diff
@@ -134,46 +127,109 @@ impl PayableDao for PayableDaoReal {
+                wallets_and_rowids.iter().map(|(_, rowid)|rowid.to_string()).join(COMMA_SEPARATOR),
+                 comma_separated_wallets(wallets_and_rowids,"'")
+            );
+            let failing_wallets = conn
+                .prepare(&sql)
+                .expect("select failed")
+                .query_map([], |row| row.get::<usize, String>(0))
+                .expect("no args but yet binding failed")
+                .flatten() //TODO use vigilant flatten
+                .join(COMMA_SEPARATOR);
+            if failing_wallets.is_empty() {
+                failing_wallets
+            } else {
+                format!(" Accounts for wallets ({}) had contained rowids when we tried to update them with new ones. \
+                 All such columns should be emptied beforehand at the confirmation of the earlier transactions. \
+                 A malformed, repeated payment is suspected", failing_wallets)
+            }
         }
+        fn comma_separated_wallets(
+            wallets_and_rowids: &[(&Wallet, u64)],
+            quoting_mark: &str,
+        ) -> String {
+            wallets_and_rowids
+                .iter()
+                .map(|(wallet, _)| format!("{quoting_mark}{wallet}{quoting_mark}"))
+                .join(COMMA_SEPARATOR)
+        }
+        fn resolve_success_or_failure(
+            conn: &dyn ConnectionWrapper,
+            wallets_and_rowids: &[(&Wallet, u64)],
+            returning_clause_feedback: Result<
+                impl Iterator<Item = Result<Option<()>, rusqlite::Error>>,
+                rusqlite::Error,
+            >,
+        ) -> Result<(), PayableDaoError> {
+            fn rows_changed_counter(takes: Vec<Option<()>>) -> usize {
+                takes.iter().flatten().count()
+            }
+            match multi_row_update_rows_changed(returning_clause_feedback, rows_changed_counter) {
+                Ok(rows_affected) => match rows_affected {
+                    num if num == wallets_and_rowids.len() => Ok(()),
+                    num => panic!(
+                        "Marking pending payable rowid for wallets {} affected {} rows but expected {}.{}",
+                        comma_separated_wallets(wallets_and_rowids,""),
+                        num,
+                        wallets_and_rowids.len(),
+                        error_extension_about_not_null_ids(conn, wallets_and_rowids)
+                    ),
+                },
+                Err(e) => Err(PayableDaoError::RusqliteError(e.to_string())),
+            }
+        }
+
+        let sql = {
+            let when_clauses_for_case_stm = wallets_and_rowids
+                .iter()
+                .map(|(wallet, rowid)| format!("when wallet_address = '{}' then {}", wallet, rowid))
+                .join("\n");
+            format!(
+                "update payable set pending_payable_rowid = case {} end \
+             where pending_payable_rowid is null
```

Comment:
Maybe extend this WHERE clause with `and wallet_address in ()` with a list of wallet addresses? That means potentially a much smaller number of records would be updated.